test/e2e/conftest.py

```diff
@@ -0,0 +1,299 @@
+        KEEP_WORKER_AFTER_FAILURE: If set to "true", will not destroy the Worker when it fails. Useful for debugging. Default is "false"
+
+    Returns:
+        DeadlineWorker: Instance of the DeadlineWorker class that can be used to interact with the Worker.
+    """
+
+    worker: DeadlineWorker
+    if os.environ.get("USE_DOCKER_WORKER", "").lower() == "true":
+        LOG.info("Creating Docker worker")
+        worker = DockerContainerWorker(
+            configuration=worker_config,
+        )
+    else:
+        LOG.info("Creating EC2 worker")
+        ami_id = os.getenv("AMI_ID")
+        subnet_id = os.getenv("SUBNET_ID")
+        security_group_id = os.getenv("SECURITY_GROUP_ID")
+        assert subnet_id, "SUBNET_ID is required when deploying an EC2 worker"
+        assert security_group_id, "SECURITY_GROUP_ID is required when deploying an EC2 worker"
+
+        bootstrap_resources: BootstrapResources = request.getfixturevalue("bootstrap_resources")
+        assert (
+            bootstrap_resources.worker_instance_profile_name
+        ), "Worker instance profile is required when deploying an EC2 worker"
+
+        ec2_client = boto3.client("ec2")
+        s3_client = boto3.client("s3")
+        ssm_client = boto3.client("ssm")
+
+        worker = EC2InstanceWorker(
+            ec2_client=ec2_client,
+            s3_client=s3_client,
+            bootstrap_bucket_name=bootstrap_resources.bootstrap_bucket_name,
+            ssm_client=ssm_client,
+            override_ami_id=ami_id,
+            subnet_id=subnet_id,
+            security_group_id=security_group_id,
+            instance_profile_name=bootstrap_resources.worker_instance_profile_name,
+            configuration=worker_config,
+        )
+
+    def stop_worker():
+        if request.session.testsfailed > 0:
+            if os.getenv("KEEP_WORKER_AFTER_FAILURE", "false").lower() == "true":
+                LOG.info("KEEP_WORKER_AFTER_FAILURE is set, not stopping worker")
+                return
+
+        try:
+            worker.stop()
+        except Exception as e:
+            LOG.exception(f"Error while stopping worker: {e}")
+            LOG.error(
+                "Failed to stop worker. Resources may be left over that need to be cleaned up manually."
+            )
+            raise
+
+    try:
+        worker.start()
+    except Exception as e:
+        LOG.exception(f"Failed to start worker: {e}")
+        LOG.info("Stopping worker because it failed to start")
+        stop_worker()
+        raise
+
+    yield worker
+
+    stop_worker()
+
+
+@pytest.fixture(scope="session")
+def region() -> str:
+    return os.getenv("REGION", os.getenv("AWS_DEFAULT_REGION", "us-west-2"))
+
+
+@pytest.fixture(scope="session")
+def job_run_as_user() -> PosixSessionUser:
+    return PosixSessionUser(
+        user="jobuser",
+        group="shared-group",
+    )
+
+
+@pytest.fixture(scope="session")
+def operating_system(request) -> str:
+    if request.param == "linux":
```

Comment:
This fixture is not parametrized. Should this be:

```suggestion
@pytest.fixture(scope="session", params=("linux", "windows"))
def operating_system(request) -> str:
    if request.param == "linux":
```

Reply:
Updated